### ISSUE TITLE
Unreasonably high ghostscript resolution takes a long time

### ISSUE THREAD

**Issue by izahn** [ocrmypdf/OCRmyPDF#981]
**Describe the bug**

Attempting to OCR https://www.accessdata.fda.gov/cdrh_docs/pdf16/K161141.pdf takes a very long time. I think this is because `gs` gets called with very high resolution argument.

**To Reproduce**

```bash
ocrmypdf --skip-text -j 1 -v 2 ~/Downloads/K161141.pdf test.pdf
```
The out put looks like this:

```
  DEBUG ocrmypdf - ocrmypdf 13.4.7
  DEBUG ocrmypdf.subprocess - Running: ['tesseract', '--list-langs']
  DEBUG ocrmypdf.subprocess.tesseract - stdout/stderr = List of available languages in "/usr/share/tessdata/" (2):
eng
osd

  DEBUG ocrmypdf.subprocess - Running: ['tesseract', '--version']
  DEBUG ocrmypdf.subprocess - Found tesseract 5.1.0
  DEBUG ocrmypdf.subprocess - Running: ['gs', '--version']
  DEBUG ocrmypdf.subprocess - Found gs 9.56.1
  DEBUG ocrmypdf.helpers - os.symlink(/home/izahn/Downloads/K161141 (1).pdf, /tmp/ocrmypdf.io.2l8pr1ms/origin)
  DEBUG ocrmypdf.helpers - os.symlink(/tmp/ocrmypdf.io.2l8pr1ms/origin, /tmp/ocrmypdf.io.2l8pr1ms/origin.pdf)
Scanning contents: 100%|███████████████████████████████████████████████████████████████████████████████| 10/10 [00:06<00:00,  1.50page/s]
  DEBUG ocrmypdf.builtin_plugins.tesseract_ocr - Using Tesseract OpenMP thread limit 1
   INFO ocrmypdf._pipeline -    1  skipping all processing on this page                                                                  
   INFO ocrmypdf._pipeline -    2  skipping all processing on this page                                                                  
  DEBUG ocrmypdf._graft -    1  Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0                         
   INFO ocrmypdf._pipeline -    3  skipping all processing on this page                                                                  
  DEBUG ocrmypdf._graft -    1  Page rotation: (content, auto) -> page = (0, 0) -> 0                                                     
  DEBUG ocrmypdf._pipeline -    4  Rasterize with png16m, rotation 0                                                                     
  DEBUG ocrmypdf._graft -    2  Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0                         
  DEBUG ocrmypdf.subprocess -    4  Running: ['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dInterpolateControl=-1', '-sDEVICE=png16m', '-dFirstPage=4', '-dLastPage=4', '-r33000.000000x33000.000000', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io.2l8pr1ms/origin.pdf']
  DEBUG ocrmypdf._graft -    2  Page rotation: (content, auto) -> page = (0, 0) -> 0                                                     
  DEBUG ocrmypdf._graft -    3  Text rotation: (text, autorotate, content) -> text misalignment = (0, 0, 0) -> 0                         
  DEBUG ocrmypdf._graft -    3  Page rotation: (content, auto) -> page = (0, 0) -> 0                                                     
OCR:  25%|██████████████████████▌                                                                   | 2.5/10.0 [00:00<00:00, 96.56page/s]
```
and stays there at 25% for at least 20 minutes (I eventually killed it).

My guess is that `-r33000.000000x33000.000000` is the problem here, as that resolution looks impossibly high to me. That's just my guess though.

**Example file**

I've only seen this problem on this file:
https://www.accessdata.fda.gov/cdrh_docs/pdf16/K161141.pdf

**System**
 - OS: Arch LInux
 - OCRmyPDF Version: 13.4.7. 
 - I've replicated this with ocrmypdf from the Archlinux package and also from conda-forge, with the same result.

**Comment by jbarlow83** [comment]
It seems there's an issue with calculating the DPI requirement for inline images. The logo in this PDF is constructed from about 3000 inline images (which is odd, to say the least, but allowed).

In the interest of getting this file OCR'd before I release a fix, you could do something like:

```
qpdf --ii-min-bytes=0 --externalize-inline-images input.pdf step.pdf
ocrmypdf -s -v2 step.pdf out.pdf
```
i.e. convert all inline images to regular, and then let ocrmypdf sort it out.

This file is also a good candidate for using `--force-ocr` to clean up its unusual formatting.

**Comment by jbarlow83** [comment]
This should be fixed (properly) by upgrading to pikepdf 5.3.1 when it is released (shortly). Due to a problem with the inline image parser, ocrmypdf as not seeing all of the images that were actually present.

See
https://github.com/pikepdf/pikepdf/issues/299
